### PR TITLE
doc: Added Thingy:91 X sections under Developing with nRF91

### PR DESCRIPTION
**Pull request by divipillai**
Adding Thingy:91 X sections under Developing
with nRF91 Series devices.
NCSDK-29616

### CODE REVIEW

File: doc/nrf/app_dev/device_guides/nrf91/thingy91x_updating_fw_programmer.rst [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+         .. code-block:: console
+
+            nrfutil device program --firmware <name_of_network_core_binary.hex> --serial-number <J-Link Serial number> --traits jlink --x-family nrf53 --core Network
+
+.. _update_nRF9151_application:
+
+Updating the application firmware on the nRF9151 SiP
+****************************************************
+
+The following section describes how the application firmware of the nRF9151 SiP on the Nordic Thingy:91 X can be updated through USB or with an external debug probe.
```
Comment (peknis) [inline doc/nrf/app_dev/device_guides/nrf91/thingy91x_updating_fw_programmer.rst:92]:
```suggestion
This section describes how you can update the application firmware of the nRF9151 SiP on the Nordic Thingy:91 X through USB or with an external debug probe.
```

File: doc/nrf/app_dev/device_guides/nrf91/thingy91x_updating_fw_programmer.rst [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+.. tabs::
+
+   .. group-tab:: Through USB and MCUboot
+
+      To update the nRF5340 SoC firmware over USB, complete the following steps:
+
+      1. Install the nrfutil device command package by completing the steps in the `Installing and upgrading nRF Util commands`_ documentation.
```
Comment (peknis) [inline doc/nrf/app_dev/device_guides/nrf91/thingy91x_updating_fw_programmer.rst:100]:
```suggestion
      1. Install the ``nrfutil device`` command package by completing the steps in the `Installing and upgrading nRF Util commands`_ documentation.
```

File: doc/nrf/app_dev/device_guides/nrf91/thingy91x_updating_fw_programmer.rst [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+      #. Enter the following command to list the connected devices and their traits::
+
+          nrfutil device list
+
+         The Nordic Thingy\:91 X will be listed as a Thingy\:91 X UART product and have following details:
+
+         * A 21 character serial number, for example, ``THINGY91X_C2E0AC7F599``.
+         * ``mcuboot``, ``nordicUsb``, ``serialPorts``, and ``usb`` traits.
+
+      #. Enter the following command to program an application binary to the nRF9151 application core:
```
Comment (peknis) [inline doc/nrf/app_dev/device_guides/nrf91/thingy91x_updating_fw_programmer.rst:113]:
```suggestion
      #. Enter the following command to program the application binary to the nRF9151 application core:
```

File: doc/nrf/app_dev/device_guides/nrf91/thingy91x_updating_fw_programmer.rst [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+
+         .. code-block:: console
+
+            nrfutil device program --firmware dfu_application.zip --serial-number <J-Link Serial number> --traits mcuboot --x-family nrf91 --core Application
+
+   .. group-tab:: Through external debug probe
+
+      To update the nRF5340 firmware using an external debug probe, complete the following steps:
+
+      1. Install the nrfutil device command package by completing the steps in the `Installing and upgrading nRF Util commands`_ documentation.
```
Comment (peknis) [inline doc/nrf/app_dev/device_guides/nrf91/thingy91x_updating_fw_programmer.rst:123]:
```suggestion
      1. Install the ``nrfutil device`` command package by completing the steps in the `Installing and upgrading nRF Util commands`_ documentation.
```

File: doc/nrf/app_dev/device_guides/nrf91/thingy91x_updating_fw_programmer.rst [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+
+The following section describes how the firmware of the nRF5340 SoC on the Nordic Thingy:91 X can be updated through USB or with an external debug probe.
+
+.. tabs::
+
+   .. group-tab:: Through USB and MCUboot
+
+      To update the nRF5340 SoC firmware over USB, complete the following steps:
+
+      1. Install the nrfutil device command package by completing the steps in the `Installing and upgrading nRF Util commands`_ documentation.
```
Comment (peknis) [inline doc/nrf/app_dev/device_guides/nrf91/thingy91x_updating_fw_programmer.rst:38]:
```suggestion
      1. Install the ``nrfutil device`` command package by completing the steps in the `Installing and upgrading nRF Util commands`_ documentation.
```

File: doc/nrf/app_dev/device_guides/nrf91/thingy91x_updating_fw_programmer.rst [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+      #. Power on the device by switching **SW1** to the **ON** position.
+      #. Set the programming target select switch **SW2** to **nRF91** on the Thingy:91 X.
+      #. Open a terminal window.
+      #. Enter the following command to list the connected devices and their traits::
+
+          nrfutil device list
+
+         The external debug probe will be listed as a J-Link product with the ``jlink`` trait and will have a 9 or 10 digit J-Link serial number depending on the J-Link probe used.
+
+      #. Enter the following command to program an application binary to the nRF9151 application core:
```
Comment (peknis) [inline doc/nrf/app_dev/device_guides/nrf91/thingy91x_updating_fw_programmer.rst:136]:
```suggestion
      #. Enter the following command to program the application binary to the nRF9151 application core:
```

File: doc/nrf/app_dev/device_guides/nrf91/thingy91x_updating_fw_programmer.rst [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+
+Updating the modem firmware on the nRF9151 SiP
+**********************************************
+
+.. note::
+   Modem firmware update through USB and MCUboot is currently not supported.
+
+To update the nRF5340 firmware using an external debug probe, complete the following steps:
+
+1. Install the nrfutil device command package by completing the steps in the `Installing and upgrading nRF Util commands`_ documentation.
```
Comment (peknis) [inline doc/nrf/app_dev/device_guides/nrf91/thingy91x_updating_fw_programmer.rst:152]:
```suggestion
1. Install the ``nrfutil device`` command package by completing the steps in the `Installing and upgrading nRF Util commands`_ documentation.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by bjda — APPROVED** [review]

File: doc/nrf/app_dev/device_guides/nrf91/thingy91x_updating_fw_programmer.rst [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+
+      1. Install the nrfutil device command package by completing the steps in the `Installing and upgrading nRF Util commands`_ documentation.
+      #. Connect the Thingy:91 X to your computer with a USB-C cable.
+      #. Power on the device by switching **SW1** to the **ON** position.
+      #. Open a terminal window.
+      #. Enter the following command to list the connected devices and their traits::
+
+          nrfutil device list
+
+         The Nordic Thingy\:91 X will be listed as a Thingy\:91 X UART product and have following details:
```
Comment (richabp) [inline doc/nrf/app_dev/device_guides/nrf91/thingy91x_updating_fw_programmer.rst:46]:
```suggestion
         The Nordic Thingy\:91 X will be listed as a Thingy\:91 X UART product and have the following details:
```

File: doc/nrf/app_dev/device_guides/nrf91/thingy91x_updating_fw_programmer.rst [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+
+         * A 21 character serial number, for example, ``THINGY91X_C2E0AC7F599``.
```
Comment (richabp) [inline doc/nrf/app_dev/device_guides/nrf91/thingy91x_updating_fw_programmer.rst:48]:
```suggestion
         * A 21-character serial number, for example, ``THINGY91X_C2E0AC7F599``.
```

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 10]
```diff
@@ -111,6 +111,7 @@ Developing with nRF91 Series
 
   * The :ref:`nRF91 modem tracing with RTT backend snippet <nrf91_modem_trace_rtt_snippet>` to enable modem tracing using the RTT trace backend.
   * The :ref:`nRF91 modem tracing with RAM backend snippet <nrf91_modem_trace_ram_snippet>` to enable modem tracing using the RAM trace backend.
+  * Guides for Thingy:91 X on the :ref:`ug_nrf91` documentation.
```
Comment (umapraseeda) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:114]:
```suggestion
  * User guide for Thingy:91 X in the :ref:`ug_nrf91` documentation.
```

File: doc/nrf/app_dev/device_guides/nrf91/thingy91x_updating_fw_programmer.rst [hunk 11]
```diff
@@ -0,0 +1,169 @@
...
+
+      1. Install the nrfutil device command package by completing the steps in the `Installing and upgrading nRF Util commands`_ documentation.
+      #. Connect the Thingy:91 X to your computer with a USB-C cable.
+      #. Power on the device by switching **SW1** to the **ON** position.
+      #. Open a terminal window.
+      #. Enter the following command to list the connected devices and their traits::
+
+          nrfutil device list
+
+         The Nordic Thingy\:91 X will be listed as a Thingy\:91 X UART product and have following details:
```
Comment (richabp) [inline doc/nrf/app_dev/device_guides/nrf91/thingy91x_updating_fw_programmer.rst:108]:
```suggestion
         The Nordic Thingy\:91 X will be listed as a Thingy\:91 X UART product and have the following details:
```

File: doc/nrf/app_dev/device_guides/nrf91/thingy91x_updating_fw_programmer.rst [hunk 12]
```diff
@@ -0,0 +1,169 @@
...
+
+         * A 21 character serial number, for example, ``THINGY91X_C2E0AC7F599``.
```
Comment (richabp) [inline doc/nrf/app_dev/device_guides/nrf91/thingy91x_updating_fw_programmer.rst:110]:
```suggestion
         * A 21-character serial number, for example, ``THINGY91X_C2E0AC7F599``.
```

File: doc/nrf/external_comp/nrf_cloud.rst [hunk 13]
```diff
@@ -73,7 +73,7 @@ nRF Cloud supports a variety of device ID formats:
 
 Nordic development kits such as the nRF9160 DK have a sticker on the back indicating the device's IMEI and PIN.
 They are preprovisioned with just-in-time provisioning (JITP) certificates so they can be added easily to nRF Cloud using the device ID ``nrf-IMEI`` and PIN.
-Device IDs that start with ``nrf-`` are checked against a database of genuine Nordic development kits and Thingy:91s.
+Device IDs that start with ``nrf-`` are checked against a database of genuine Nordic development kits and Thingy:91 and Thingy:91 X.
```
Comment (umapraseeda) [inline doc/nrf/external_comp/nrf_cloud.rst:76]:
```suggestion
Device IDs that start with ``nrf-`` are checked against a database of genuine Nordic development kits, Thingy:91, and Thingy:91 X.
```

**Review by richabp — CHANGES_REQUESTED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 10 above)
Comment (divipillai) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst]: Removed from this PR and added this comment to the 2.8 release notes PR.

**Review by peknis — APPROVED** [review]

File: doc/nrf/external_comp/memfault.rst [hunk 14]
```diff
@@ -202,7 +202,7 @@ The following samples demonstrate the Memfault integration in |NCS|:
 
 * :ref:`peripheral_mds` - This sample has an implementation of the Memfault Diagnostic Service (MDS).
   It sends data to the Memfault cloud through a Bluetooth gateway.
-* :ref:`memfault_sample` - This sample connects to an LTE network using an nRF91 Series DK or Thingy:91, or to Wi-Fi using the nRF7002 DK, and sends the collected data to Memfault's cloud using HTTPS.
+* :ref:`memfault_sample` - This sample connects to an LTE network using an nRF91 Series DK, Thingy:91, or Thingy:91 X, or to Wi-Fi using the nRF7002 DK, and sends the collected data to Memfault's cloud using HTTPS.
```
Comment (umapraseeda) [inline doc/nrf/external_comp/memfault.rst:205]:
```suggestion
* :ref:`memfault_sample` - This sample connects to an LTE network using an nRF91 Series DK, Thingy:91, or Thingy:91 X, or to Wi-Fi® using the nRF7002 DK, and sends the collected data to Memfault's cloud using HTTPS.
```

**Review by umapraseeda — APPROVED** [review]

File: doc/nrf/app_dev/device_guides/nrf91/nrf91_programming.rst [hunk 15]
```diff
@@ -111,3 +121,54 @@ Complete the following steps to program firmware onto Thingy:91:
...
+|:file:`app_update.bin` | MCUboot compatible image, binary format|* Using the built-in bootloader and mcumgr command-line tool.   |
+|                       |                                        |* For FOTA updates.                                             |
++-----------------------+----------------------------------------+----------------------------------------------------------------+
+
+Complete the following steps to program firmware onto Thingy:91 X:
+
+1. Set the Thingy:91 X SWD selection switch (**SW2**) to **nRF91** or **nRF53** depending on whether you want to program the nRF9151 SiP or the nRF5340 SoC component.
+#. Connect the Thingy:91 X to the debug out port on a 10-pin external debug probe, for example, nRF9151 DK, using a 10-pin SWD cable.
+#. Connect the external debug probe to the PC using a USB cable.
+#. Make sure that the Thingy:91 and the external debug probe are powered on.
```
Comment (bjornspock) [inline doc/nrf/app_dev/device_guides/nrf91/nrf91_programming.rst:170]: Correct to "Thingy:91 X"

**Review by bjornspock — CHANGES_REQUESTED** [review]

**Review by bjornspock — APPROVED** [review]

File: doc/nrf/app_dev/device_guides/nrf91/thingy91x_updating_fw_programmer.rst [hunk 16]
```diff
@@ -0,0 +1,169 @@
+.. _programming_thingy91x:
+
+Updating the Thingy:91 X firmware using nrfutil device
+######################################################
```
[the remark this answers is not in the document]
Comment (bjda) [reply doc/nrf/app_dev/device_guides/nrf91/thingy91x_updating_fw_programmer.rst]: "nrfutil" and "nrfutil device" both look entirely fine to me. Both are correct, and "nrfutil device" is more precise.

**Review by richabp — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**